crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -311,63 +316,64 @@ where
+            .unwrap_or_default())
     }
 
     #[cfg(test)]
     pub fn get_next_object_version(&self, obj: &ObjectID) -> Option<SequenceNumber> {
-        self.tables.next_object_versions.get(obj).unwrap()
+        self.tables.next_shared_object_versions.get(obj).unwrap()
     }
 
     /// Read a lock for a specific (transaction, shared object) pair.
     #[cfg(test)] // Nothing wrong with this function, but it is not currently used outside of tests
-    pub fn get_assigned_object_versions<'a>(
+    pub fn get_assigned_shared_object_versions<'a>(
         &self,
         transaction_digest: &TransactionDigest,
         object_ids: impl Iterator<Item = &'a ObjectID>,
     ) -> Result<Vec<Option<SequenceNumber>>, SuiError> {
-        let keys = object_ids.map(|objid| (*transaction_digest, *objid));
-
-        self.tables
-            .assigned_object_versions
-            .multi_get(keys)
-            .map_err(SuiError::from)
+        let object_keys: std::collections::BTreeMap<_, _> = self
+            .get_shared_locks(transaction_digest)?
+            .into_iter()
+            .collect();
+        Ok(object_ids.map(|id| object_keys.get(id).cloned()).collect())
     }
 
-    pub fn remove_shared_objects_locks(
+    pub fn delete_shared_object_versions(
         &self,
-        sequenced_to_delete: &[(TransactionDigest, ObjectID)],
-        schedule_to_delete: &[ObjectID],
+        executed_transaction: &TransactionDigest,
+        deleted_objects: &[ObjectID],
     ) -> SuiResult {
-        let mut write_batch = self.tables.assigned_object_versions.batch();
-        write_batch =
-            write_batch.delete_batch(&self.tables.assigned_object_versions, sequenced_to_delete)?;
+        let mut write_batch = self.tables.assigned_shared_object_versions.batch();
+        write_batch = write_batch.delete_batch(
+            &self.tables.assigned_shared_object_versions,
+            vec![executed_transaction],
+        )?;
         write_batch =
-            write_batch.delete_batch(&self.tables.next_object_versions, schedule_to_delete)?;
+            write_batch.delete_batch(&self.tables.next_shared_object_versions, deleted_objects)?;
         write_batch.write()?;
         Ok(())
     }
 
-    pub fn insert_assigned_shared_object_versions(
+    pub fn set_assigned_shared_object_versions(
         &self,
-        sequenced: Vec<((TransactionDigest, ObjectID), SequenceNumber)>,
+        transaction_digest: &TransactionDigest,
+        assigned_versions: Vec<(ObjectID, SequenceNumber)>,
     ) -> SuiResult {
-        let mut write_batch = self.tables.assigned_object_versions.batch();
-        write_batch = write_batch.insert_batch(&self.tables.assigned_object_versions, sequenced)?;
+        let mut write_batch = self.tables.assigned_shared_object_versions.batch();
+        write_batch = write_batch.insert_batch(
+            &self.tables.assigned_shared_object_versions,
+            vec![(transaction_digest, assigned_versions)],
```

Comment:
you could either use `std::iter::once` here or not do a batch at all, since its a single write.